app/src/common/shared/org/mozilla/vrbrowser/utils/UrlUtils.java

```diff
@@ -0,0 +1,160 @@
+    }
+
+    public static String stripUserInfo(@Nullable String url) {
+        if (TextUtils.isEmpty(url)) {
+            return "";
+        }
+
+        try {
+            URI uri = new URI(url);
+
+            final String userInfo = uri.getUserInfo();
+            if (userInfo == null) {
+                return url;
+            }
+
+            // Strip the userInfo to minimise spoofing ability. This only affects what's shown
+            // during browsing, this information isn't used when we start editing the URL:
+            uri = new URI(uri.getScheme(), null, uri.getHost(), uri.getPort(), uri.getPath(), uri.getQuery(), uri.getFragment());
+
+            return uri.toString();
+        } catch (URISyntaxException e) {
+            // We might be trying to display a user-entered URL (which could plausibly contain errors),
+            // in this case its safe to just return the raw input.
+            // There are also some special cases that URI can't handle, such as "http:" by itself.
+            return url;
+        }
+    }
+
+    public static boolean isPermittedResourceProtocol(@Nullable final String scheme) {
+        return scheme != null && (
+                scheme.startsWith("http") ||
+                        scheme.startsWith("https") ||
+                        scheme.startsWith("file") ||
+                        scheme.startsWith("data") ||
+                        scheme.startsWith("javascript") ||
+                        scheme.startsWith("about"));
+    }
+
+    public static boolean isSupportedProtocol(@Nullable final String scheme) {
+        return scheme != null && (isPermittedResourceProtocol(scheme) || scheme.startsWith("error"));
+    }
+
+    public static boolean isInternalErrorURL(final String url) {
+        return "data:text/html;charset=utf-8;base64,".equals(url);
+    }
+
+    /**
+     * Checks that urls are non-null and are the same aside from a trailing slash.
+     *
+     * @return true if urls are the same except for trailing slash, or if either url is null.
+     */
+    public static boolean urlsMatchExceptForTrailingSlash(final String url1, final String url2) {
+        // This is a hack to catch a NPE in issue #26.
+        if (url1 == null || url2 == null) {
+            return false;
+        }
+        int lengthDifference = url1.length() - url2.length();
+
+        if (lengthDifference == 0) {
+            // The simplest case:
+            return url1.equalsIgnoreCase(url2);
+        } else if (lengthDifference == 1) {
+            // url1 is longer:
+            return url1.charAt(url1.length() - 1) == '/' &&
+                    url1.regionMatches(true, 0, url2, 0, url2.length());
+        } else if (lengthDifference == -1) {
+            return url2.charAt(url2.length() - 1) == '/' &&
+                    url2.regionMatches(true, 0, url1, 0, url1.length());
+        }
+
+        return false;
+    }
+
+    public static String stripCommonSubdomains(@Nullable String host) {
```

Comment:
This function appears to be the only one used in this file. Please don't just copy and paste whole files into the code base just to use one function.

Reply:
I have removed unused functions.